### ISSUE TITLE
[JENKINS-62323] Role Edits Work But Jenkins Restart Required To Take Effect

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/role-strategy-plugin#602]
<p>The user interface experience is error message free.</p>

<p>/var/log/jenkins/jenkins.log has many exceptions but it's hard to tell which are relevant.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/conrad_t_pino.svg" title="conrad_t_pino's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=conrad_t_pino">conrad_t_pino</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62323" target="_blank">Role Edits Work But Jenkins Restart Required To Take Effect</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=oleg_nenashev">oleg_nenashev</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: role-strategy-plugin
<li><b>label(s)</b>: jenkins, plugin, restart
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2020-10-01T08:58:19+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 5
<li><b>imported</b>: 20251211-223321
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>The user interface experience is error message free.</p>

<p>/var/log/jenkins/jenkins.log has many exceptions but it's hard to tell which are relevant.</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Role-based Authorization Strategy 2.16<br/>
Matrix Authorization Strategy Plugin 2.6.1<br/>
Jenkins 2.222.3<br/>
openjdk version &quot;1.8.0_171&quot;<br/>
OpenJDK 64-Bit Server VM (build 25.171-b11, mixed mode)<br/>
openjdk version &quot;1.8.0_171&quot;<br/>
Debian 8.11<br/>
Linux 3.16.0-10-amd64 #1 SMP Debian 3.16.81-1 (2020-01-17)
```
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>safredaas:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62323?focusedId=396662&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-396662">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hello!</p>

<p> </p>

<p>Does anyone here have a solution for this? This bug is currently affecting our workflows.</p>

<p> </p>

<p>Thank you!</p>

<p> </p>

<p>Best Regards</p>
</pre>
</details>

</ul>
</details>

<p>Hello!</p>

<p> </p>

<p>Does anyone here have a solution for this? This bug is currently affecting our workflows.</p>

<p> </p>

<p>Thank you!</p>

<p> </p>

<p>Best Regards</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62323?focusedId=398485&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-398485">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The fix was released in 3.1. Closing as a duplicate of <a href="https://issues.jenkins.io/browse/JENKINS-61508" title="Project role only after 1 hour can be effective" class="issue-link" data-issue-key="JENKINS-61508"><del>JENKINS-61508</del></a></p>
</pre>
</details>

</ul>
</details>

<p>The fix was released in 3.1. Closing as a duplicate of <a href="https://issue-redirect.jenkins.io/issue/61508" title="Project role only after 1 hour can be effective" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/61508">JENKINS-61508</a>"><del>JENKINS-61508</del></a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/conrad_t_pino.svg" title="conrad_t_pino's avatar" /> conrad_t_pino:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62323?focusedId=417225&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-417225">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thank you <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=oleg_nenashev" class="user-hover" rel="oleg_nenashev">oleg_nenashev</a></p>
</pre>
</details>

</ul>
</details>

<p>Thank you <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=oleg_nenashev" class="user-hover" rel="oleg_nenashev">oleg_nenashev</a></p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `duplicates` from Jira: <a class="jira-relationship" href="https://issue-redirect.jenkins.io/issue/61508">JENKINS-61508</a>]</i>
